File: src/bolum01/1.6/kod-dosyalari/nan_ornegi.c
#include <stdio.h>
#include <math.h>

int main(void) {
    double x = 0.0 / 0.0;                 /* NaN uretir */
    printf("x = %f\n", x);                /* "x = nan" */
    printf("x == x: %d\n", x == x);       /* 0 (yanlis!) */
    printf("isnan(x): %d\n", isnan(x));   /* 1 (dogru yontem) */
    return 0;
}

File: src/bolum01/1.6/kod-dosyalari/NanOrnegi.java
public class NanOrnegi {
    public static void main(String[] args) {
        double x = 0.0 / 0.0;                                  // NaN uretir
        System.out.println("x = " + x);                        // "x = NaN"
        System.out.println("x == x: " + (x == x));             // false
        System.out.println("Double.isNaN(x): " + Double.isNaN(x)); // true
    }
}
